### DISCUSSION TITLE
How to implement Read-Aside Cache

### DISCUSSION THREAD

**Discussion by wasiqaftab** [binaryk/laravel-restify#521] [Q&A]
Hello

Can you please guide me how i can add the cache strategies  and which files i can use or where i can write code if you can tell me little example that will be great

**Comment by wasiqaftab** [comment]
@arthurkirkosa can you please guide me if i need to implement cache what is the best way to do so

**Reply by arthurkirkosa** [reply]
Can you give me a little bit more context on what you have and what you need to achieve?

**Reply by wasiqaftab** [reply]
@arthurkirkosa Thanks for your reply. actually i have a big data set about 1 million records and i want to cache record so run some cache warmup so every hour cache will be recreated using command. what i am doing currently 
`  $cacheKey = 'forms:'.md5(json_encode(request()->all()));
        $minutes = 60;
        return Cache::remember($cacheKey, $minutes, function() use ($request) {
            return parent::index($request);
        });`

this code in index what i want if there is a way that it can cache response for all repository rather than i do that on every repository

**Reply by arthurkirkosa** [reply]
One way to do it is to create a "parent" Repository class that all "children" would extend and add the logic there.

How many records are you returning per page on a request?

**Reply by wasiqaftab** [reply]
Thanks i am returning about 500 per page. Can u give me a idea or sudo how i can attach parent repository.

**Reply by arthurkirkosa** [reply]
```
// app/Restify/Repository.php

<?php

namespace App\Restify;

use Binaryk\LaravelRestify\Http\Requests\RestifyRequest;
use Binaryk\LaravelRestify\Repositories\Repository as RestifyRepository;

abstract class Repository extends RestifyRepository
{
    ...
    
    public static function indexQuery(RestifyRequest $request, $query): void
    {
        //
    }

    ...
}


// app/Restify/EmployeeRepository.php

<?php

namespace App\Restify\Payroll;

class EmployeeRepository extends Repository
{
    public static $model = Employee::class;
    ...
}
```

Something like this

**Reply by wasiqaftab** [reply]
Ok thank you so much. Let me try that

**Reply by arthurkirkosa** [reply]
How long does it take to serve 500 records?

**Reply by wasiqaftab** [reply]
On local docker its 1s and on staging server 0.5. Because some calculations are involved in the query like radius

**Reply by arthurkirkosa** [reply]
Can you do those calculations and save them in DB as well? So you don't have to do them on GET

**Reply by wasiqaftab** [reply]
Actually its more like a property app need to show property arpund 2 or 3 mile radius. Its based on user postcode i dont find a way how i can do that thats why i am doing it on run time

**Reply by arthurkirkosa** [reply]
I see. You first need to determine the poligon / circle of coordinates representing the 2-3 mile radius and then query the DB for all properties with coordinates in that polygon; Correct?

And are you caching the polygon?

**Reply by wasiqaftab** [reply]
Yes thats what i am doing. No i am not caching polygon

**Reply by wasiqaftab** [reply]
How i can cache that

**Reply by arthurkirkosa** [reply]
You might not need to cache the polygon if it takes a few ms to compute.

Is you SQL query something like this one?

```
SELECT * FROM stuff 
 WHERE Contains(
 GeomFromText('POLYGON((41.000497 -109.050149, 41.002380 -102.051881, 36.993237 -102.041959, 36.999037 -109.045220, 41.000497 -109.050149))'), location );
```

**Reply by wasiqaftab** [reply]
Yes its like that sould i do laravel cache for that query

**Reply by arthurkirkosa** [reply]
I'm not sure you can do that, as the position of the user's coordinates change "constantly".

Maybe do a +- 0.5 mi buffer. Something like. If the radius is 3mi, do a 3.5mi polygon. And cache the result; for a change in position for the user of .5mi in any direction from the initial cache coords of the user and serve the same DB results for the initial 3.5mi polygon.

**Reply by wasiqaftab** [reply]
OK thanks thats a good idea it will work
